src/main/java/com/exasol/adapter/dynamodb/queryplanning/RemoteTableQueryFactory.java

```diff
@@ -68,20 +68,30 @@ public Void visit(final SqlSelectList selectList) {
             } else if (selectList.isSelectStar()) {
                 selectAllColumns();
             } else {
-                throw new UnsupportedOperationException(
-                        "The current version of DynamoDB Virtual Schema does not support projection.");
+                for (final SqlNode selectListExpression : selectList.getExpressions()) {
+                    if (!(selectListExpression instanceof SqlColumn)) {
+                        throw new UnsupportedOperationException(
+                                "The current version of Document Virtual Schema does not support SQL functions.");
+                    }
```

Comment:
Is it okay to throw here?
Can a next selectExpression be instance of SqlColumn?

Reply:
Yes throwing is intended here. A function should never occur as the corresponding capability is not set. If it happens an abort is intended